libs/test/SalesforceSDKTest/src/com/salesforce/androidsdk/util/TestDataCleanupHelper.java

```diff
@@ -0,0 +1,96 @@
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+package com.salesforce.androidsdk.util;
+
+import android.test.InstrumentationTestCase;
+
+import com.salesforce.androidsdk.TestCredentials;
+import com.salesforce.androidsdk.auth.HttpAccess;
+import com.salesforce.androidsdk.auth.OAuth2;
+import com.salesforce.androidsdk.auth.OAuth2.TokenEndpointResponse;
+import com.salesforce.androidsdk.rest.RestClient;
+import com.salesforce.androidsdk.rest.RestClient.ClientInfo;
+import com.salesforce.androidsdk.rest.RestRequest;
+import com.salesforce.androidsdk.rest.RestRequest.RestMethod;
+import com.salesforce.androidsdk.rest.RestResponse;
+
+import java.net.URI;
+/**
+ * Do org data cleanup
+ *
+ */
+public class TestDataCleanupHelper extends InstrumentationTestCase {
+
+    private ClientInfo clientInfo;
+    private HttpAccess httpAccess;
+    private RestClient restClient;
+    private String authToken;
+    public static final String TEST_FIRST_NAME = "firstName";
+    public static final String TEST_LAST_NAME = "lastName";
+    public static final String TEST_DISPLAY_NAME = "displayName";
+    public static final String TEST_EMAIL = "test@email.com";
+    public static final String TEST_PHOTO_URL = "http://some.photo.url";
+    public static final String TEST_THUMBNAIL_URL = "http://some.thumbnail.url";
+    private static final String EXPECTED_DELETE = "ExpectedDeletes";
+
+
+    @Override
+    public void setUp() throws Exception {
+        super.setUp();
+        TestCredentials.init(getInstrumentation().getContext());
+        httpAccess = new HttpAccess(null, "dummy-agent");
+        TokenEndpointResponse refreshResponse = OAuth2.refreshAuthToken(httpAccess, new URI(TestCredentials.INSTANCE_URL), TestCredentials.CLIENT_ID, TestCredentials.REFRESH_TOKEN);
+        authToken = refreshResponse.authToken;
+        clientInfo = new ClientInfo(TestCredentials.CLIENT_ID,
+                new URI(TestCredentials.INSTANCE_URL),
+                new URI(TestCredentials.LOGIN_URL),
+                new URI(TestCredentials.IDENTITY_URL),
+                TestCredentials.ACCOUNT_NAME, TestCredentials.USERNAME,
+                TestCredentials.USER_ID, TestCredentials.ORG_ID, null, null,
+                TEST_FIRST_NAME, TEST_LAST_NAME, TEST_DISPLAY_NAME, TEST_EMAIL, TEST_PHOTO_URL, TEST_THUMBNAIL_URL);
+        restClient = new RestClient(clientInfo, authToken, httpAccess, null);
+    }
+
+    /**
+     * Cleanup all records that created by test account.
+     *
+     * @throws Exception
+     */
+    public void testCleanupOrgData() throws Exception {
+        int stop = 0;
+        int expectedCount = 0;
+        String requestFormat = "/services/apexrest/Cleanup?start=%d&stop=%d";
```

Comment:
sorry I missed the whole conversation here, I agree with @bhariharan , let's not use the apex here since it's a blackbox to our customers... 
@helenren can we just use restAPI to achieve what we want here? Say, we use API to get a list of accounts/contacts? owned by the user, then delete them through rest API also?